src/common/extensionMessaging.ts

```diff
@@ -24,47 +21,19 @@ export interface MessageWithArguments {
     args: any[];
 }
 
-export interface IExtensionMessageSender {
-    sendMessage(message: ExtensionMessage, args?: any[]): Q.Promise<any>;
-}
-
-/**
- * Sends messages to the extension.
- */
-export class ExtensionMessageSender implements IExtensionMessageSender {
-
-    public sendMessage(message: ExtensionMessage, args?: any[]): Q.Promise<any> {
-        let deferred = Q.defer<any>();
-        let messageWithArguments: MessageWithArguments = { message: message, args: args };
-        let body = "";
-
-        let pipePath = HostPlatform.getExtensionPipePath();
-        let socket = net.connect(pipePath, function() {
-            let messageJson = JSON.stringify(messageWithArguments);
-            socket.write(messageJson);
-        });
-
-        socket.on("data", function(data: any) {
-            body += data;
-        });
-
-        socket.on("error", function(data: any) {
-            deferred.reject(new Error("An error ocurred while handling message: " + ExtensionMessage[message]));
-        });
+export let ErrorMarker = "vscodereactnative-error-marker";
 
-        socket.on("end", function() {
-            try {
-                if (body === ErrorMarker) {
-                    deferred.reject(new Error("An error ocurred while handling message: " + ExtensionMessage[message]));
-                } else {
-                    let responseBody: any = body ? JSON.parse(body) : null;
-                    deferred.resolve(responseBody);
-                }
-            } catch (e) {
-                deferred.reject(e);
-            }
-        });
+export class MessagingChannel {
+    constructor(private projectRootPath: string) {
```

Comment:
Is this constructor intentionally empty? We should add a comment with an explanation if yes.